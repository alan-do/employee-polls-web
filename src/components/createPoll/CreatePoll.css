.form-group {
    margin-bottom: 1.5rem;
    background-color: #f9f9f9;
    padding: 1.5rem;
    border-radius: 8px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.label {
    display: block;
    margin-bottom: 0.75rem;
    font-weight: bold;
    color: #333;
}

.input-container {
    margin-bottom: 1rem;
}

.input {
    width: 100%;
    padding: 0.75rem;
    border: 1px solid #ddd;
    border-radius: 4px;
    font-size: 1rem;
    transition: border-color 0.3s ease;
}

.input:focus {
    outline: none;
    border-color: #007bff;
    box-shadow: 0 0 0 2px rgba(0, 123, 255, 0.25);
}

.title {
    font-size: 2rem;
    font-weight: bold;
    margin-bottom: 1.5rem;
    color: #007bff;
    text-align: center;
}

.button {
    background-color: #007bff;
    color: white;
    padding: 0.75rem 1.5rem;
    border: none;
    border-radius: 4px;
    cursor: pointer;
    font-size: 1rem;
    font-weight: bold;
    transition: background-color 0.3s ease;
}

.button:hover {
    background-color: #0056b3;
}

.button-container {
    flex-direction: column;
    display: flex;
    justify-content: center;
    margin-top: 1.5rem;
    margin-bottom: 1.5rem;
    max-width: 800px;
}

.form {
    width: 100%;
    max-width: 800px;
    margin: 0 auto;
}

.error {
    color: red;
    font-weight: bold;
    text-align: center;
}