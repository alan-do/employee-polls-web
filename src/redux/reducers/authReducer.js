import { GRANTED_AUTHORIZATION, DENIED_AUTHORIZATION } from "../actions/authAction";

export default function authReducer(state = null, action) {
    switch (action.type) {
        case GRANTED_AUTHORIZATION:
            return action.authenticatedUser;
        case DENIED_AUTHORIZATION:
            return null;
        default:
            return state;
    }
}
