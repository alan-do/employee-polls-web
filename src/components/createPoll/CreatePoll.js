import { useNavigate } from "react-router-dom";
import { useState } from "react";
import { handleCreateQuestion } from "../../redux/actions/questionsAction";
import { useDispatch, useSelector } from "react-redux";
import "./CreatePoll.css";

const CreatePoll = () => {
    const dispatch = useDispatch();
    const navigate = useNavigate();

    const authenticatedUser = useSelector(state => state.authenticatedUser);
    const [optionOne, setOptionOne] = useState("");
    const [optionTwo, setOptionTwo] = useState("");
    const [error, setError] = useState("");

    const onChangeOptionOne = (e) => {
        const value = e.target.value;
        setOptionOne(value);
    };

    const onChangeOptionTwo = (e) => {
        const value = e.target.value;
        setOptionTwo(value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        if (optionOne.trim() === "" || optionTwo.trim() === "") {
            setError("Both options must be filled.");
            return;
        }
        dispatch(handleCreateQuestion(
            optionOne,
            optionTwo,
            authenticatedUser.id,
        ));
        navigate("/");
    };

    return (
        <div>
            <h1 className="title">New Poll</h1>
            <form onSubmit={handleSubmit} className="form">
                
                <div className="form-group">
                    <label htmlFor="optionOne"
                        data-testid="optionOneLabel"
                        className="label">Option One</label>
                    <div className="input-container" data-testid="optionOne">
                        <input
                            data-testid="optionOneInput"
                            value={optionOne}
                            onChange={onChangeOptionOne}
                            type="text"
                            name="optionOne"
                            id="optionOne"
                            className="input" />
                    </div>
                </div>

                <div className="form-group">
                    <label htmlFor="optionTwo"
                        data-testid="optionTwoLabel"
                        className="label">Option Two</label>
                    <div className="input-container" data-testid="optionTwo">
                        <input
                            data-testid="optionTwoInput"
                            value={optionTwo}
                            onChange={onChangeOptionTwo}
                            type="text"
                            name="optionTwo"
                            id="optionTwo"
                            className="input" />
                    </div>
                </div>

                <div className="button-container">
                {error && <p className="error">{error}</p>}
                    <button type="submit"
                        data-testid="submit-poll"
                        className="button">
                        Submit
                    </button>
                </div>

            </form>
        </div>
    );
};

export default CreatePoll;