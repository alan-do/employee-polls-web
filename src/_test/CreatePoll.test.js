import { fireEvent, render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import React from "react";
import CreatePoll from "../components/createPoll/CreatePoll";
import { configureStore } from "@reduxjs/toolkit";
import authReducer from "../redux/reducers/authReducer";
import questionsReducer from "../redux/reducers/questionReducer";
import usersReducer from "../redux/reducers/userReducer";

const createMockStore = (initialState) => {
    return configureStore({
        reducer: {
            authenticatedUser: authReducer,
            questions: questionsReducer,
            users: usersReducer,
        },
        preloadedState: initialState,
    });
};

describe("CreatePoll", () => {
    it("should render the component", () => {
        const view = render(
            <Provider store={createMockStore({})}>
                <Router>
                    <CreatePoll/>
                </Router>
            </Provider>
        );
        expect(view).toBeDefined();
        expect(view).toMatchSnapshot();
    });

    it("should display all elements", () => {
        render(
            <Provider store={createMockStore({})}>
                <Router>
                    <CreatePoll/>
                </Router>
            </Provider>
        );

        const optionOneLabelElement = screen.getByTestId("optionOneLabel");
        const optionOneInputElement = screen.getByTestId("optionOneInput");
        const optionTwoLabelElement = screen.getByTestId("optionTwoLabel");
        const optionTwoInputElement = screen.getByTestId("optionTwoInput");
        const submitButtonElement = screen.getByTestId("submit-poll");

        expect(optionOneLabelElement.textContent).toBe("Option One");
        expect(optionTwoLabelElement.textContent).toBe("Option Two");
        expect(submitButtonElement.textContent).toBe("Submit");

        fireEvent.change(optionOneInputElement, {target: {value: 'rent new office'}});
        fireEvent.change(optionTwoInputElement, {target: {value: 'rent old office'}});
        
        expect(optionOneInputElement.value).toBe("rent new office");
        expect(optionTwoInputElement.value).toBe("rent old office");
    });
});
