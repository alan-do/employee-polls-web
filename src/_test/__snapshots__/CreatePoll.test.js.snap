// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePoll should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1
          class="title"
        >
          New Poll
        </h1>
        <form
          class="form"
        >
          <div
            class="form-group"
          >
            <label
              class="label"
              data-testid="optionOneLabel"
              for="optionOne"
            >
              Option One
            </label>
            <div
              class="input-container"
              data-testid="optionOne"
            >
              <input
                class="input"
                data-testid="optionOneInput"
                id="optionOne"
                name="optionOne"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="form-group"
          >
            <label
              class="label"
              data-testid="optionTwoLabel"
              for="optionTwo"
            >
              Option Two
            </label>
            <div
              class="input-container"
              data-testid="optionTwo"
            >
              <input
                class="input"
                data-testid="optionTwoInput"
                id="optionTwo"
                name="optionTwo"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="button-container"
          >
            <button
              class="button"
              data-testid="submit-poll"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1
        class="title"
      >
        New Poll
      </h1>
      <form
        class="form"
      >
        <div
          class="form-group"
        >
          <label
            class="label"
            data-testid="optionOneLabel"
            for="optionOne"
          >
            Option One
          </label>
          <div
            class="input-container"
            data-testid="optionOne"
          >
            <input
              class="input"
              data-testid="optionOneInput"
              id="optionOne"
              name="optionOne"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="form-group"
        >
          <label
            class="label"
            data-testid="optionTwoLabel"
            for="optionTwo"
          >
            Option Two
          </label>
          <div
            class="input-container"
            data-testid="optionTwo"
          >
            <input
              class="input"
              data-testid="optionTwoInput"
              id="optionTwo"
              name="optionTwo"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="button-container"
        >
          <button
            class="button"
            data-testid="submit-poll"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
